### DISCUSSION TITLE
No database dumps found in backup `SMS/2023-10-24-13-02-50.zip`.

### DISCUSSION THREAD

**Discussion by MakaraDevOP** [stefanzweifel/laravel-backup-restore#37] [Q&A] [answered]
hello, I have a problem with restoring the backup database 
i using command: php artisan backup:restore --backup=latest

error appear like this , can you help ?

![image](https://github.com/stefanzweifel/laravel-backup-restore/assets/97021587/e6289e91-1211-4012-9f02-cc67df543fc6)

**Comment by stefanzweifel** [comment]
Well, I think the exception is quite clear: Your backup seems to not contain a database dump that can be imported.

To be more exact, there doesn't seem to be a `db-dumps`-directory. Here's a direct link to the code that throws that exception: [link](https://github.com/stefanzweifel/laravel-backup-restore/blob/main/src/PendingRestore.php#L66).

Could you check how the decompressed backup looks like? You should be able to find the downloaded backup in `storage/app/backup-restore-temp`.


storage/backup-temp

**Reply by MakaraDevOP** [reply]
Thank you for your response to my questions,

I have checked the `storage/app/backup-restore-temp`, it has a temps backup that is a zip file , and contains `db-dumps` and `.sql` 
how can I fix it? 
dependencies:

       "php": "^8.0.2",
        "carlos-meneses/laravel-mpdf": "^2.1",
        "guzzlehttp/guzzle": "^7.2",
        "laravel/framework": "^9.19",
        "laravel/sanctum": "^3.0",
        "laravel/tinker": "^2.7",
        "laravelcollective/html": "^6.3",
        "maatwebsite/excel": "*",
        "mpdf/mpdf": "^8.2",
        "spatie/laravel-backup": "^8.3",
        "spatie/laravel-html": "*",
        "spatie/laravel-permission": "^5.5",
        "wnx/laravel-backup-restore": "^1.0"



![image](https://github.com/stefanzweifel/laravel-backup-restore/assets/97021587/7a337f75-88f0-4552-aff1-440cc7ca1922)

**Reply by stefanzweifel** [reply]
Ah. I see the problem.
You're on Windows. I haven't tested the package on Windows yet.

The problem is probably that I've used `/` as the directory seperator in many places. I should use `DIRECTORY_SEPARATOR` instead.

I will publish a fix in the next few days.

In the meantime, you could update [`vendor/wnx/laravel-backup-restore/src/PendingRestore.php`](
https://github.com/stefanzweifel/laravel-backup-restore/blob/main/src/PendingRestore.php) to use `\` instead of `/` in the various path declarations.

This should hopefully temporarily fix the problem with the command on your machine.

**Reply by MakaraDevOP** [reply]
I appreciate it and hope that it is fixed. Thank you.

**Reply by stefanzweifel** [reply]
@MakaraDevOP I've just released v1.1.2 with a fix. (See #38. I was unable to get my GitHub Actions to test the package an Windows, due to problems running and booting MySQL on the Windows runners)

https://github.com/stefanzweifel/laravel-backup-restore/releases/tag/v1.1.2

**Reply by MakaraDevOP** [reply]
i upgrated to v1.1.2 , i got error CLI mysql not found ,
please help me , i try to set up many project still error . 

![image](https://github.com/stefanzweifel/laravel-backup-restore/assets/97021587/17987ea5-b3b6-421d-bf5f-9a1c56521bb8)

**Reply by stefanzweifel** [reply]
Oh Windows. 🙃 

Can you tell/explain to me, how you've installed MySQL on your Windows machine?

In a recent path, I've updated the package to check if all dependencies are installed before downloading the backup.
The package is now checking if the `mysql`-binary is available to the PHP process.

I assume you don't use Linux on Windows through WSL? ([link](https://learn.microsoft.com/en-us/windows/wsl/))

On my Mac I use a tool called DBngin to install databases. I had to update my terminal profile like seen in the link below, so that `mysql` is available to the PHP process.
https://github.com/stefanzweifel/dotfiles/blob/master/zsh/zshrc.zsh#L57

**Reply by MakaraDevOP** [reply]
On Window, I'm using XAMPP, which contains PHP, MySQL 
Path of MySQL :

            'dump' => [
                'dump_binary_path' => 'D:\xampp\mysql\bin', 
                'use_single_transaction',
                'timeout' => 60 * 5, // 5 minute timeout
                'exclude_tables' => ['backup_restore'],
            ]

the package Laravel/backup has config in database.php, which points path of MySQL, i think you not use `dump_binary_path` for restoration. 

Please make sure this package supports on  window

**Reply by stefanzweifel** [reply]
@MakaraDevOP Yeah, the package currently doesn't use `database.connections.{connection}.dump.dump_binary_path`. Will create a fix in the upcoming days.

**Reply by MakaraDevOP** [reply]
@stefanzweifel  i'm waiting for you, thanks

**Reply by stefanzweifel** [reply]
I've started working on a fix in #40.
Would be great if you could give this a try by updating your project @MakaraDevOP.

**Update composer.json**

```diff
- "wnx/laravel-backup-restore": "^1.0",
+ "wnx/laravel-backup-restore": "dev-feat/use-dump_binary_path",
```

Install new version

```shell
composer upgrade wnx/laravel-backup-restore
```

Then run `php artisan backup:restore` as ususal.
As mentioned several times, I don't own a Windows device and running our test suite on GitHub Actions on Windows is quite tricky (#41). So errors can still happen.

Feel free to mention them here or better, [create an issue](https://github.com/stefanzweifel/laravel-backup-restore/issues/new).

**Reply by MakaraDevOP** [reply, marked as answer]
"wnx/laravel-backup-restore": "dev-feat/use-dump_binary_path",
    
This version supports MySQL CLI, but for my project, I need to change the path of PathToLocalDecompressedBackup to a public folder, and it's working 🎉🎉,Issue Fixed  
  
From this :

    public function getAbsolutePathToLocalDecompressedBackup(): string
    {
        $filename = $this->restoreId;

        return storage_path('app'.DIRECTORY_SEPARATOR.'backup-restore-temp'.DIRECTORY_SEPARATOR.$filename);
    }

Changed to: 👍🏼 
    
    public function getAbsolutePathToLocalDecompressedBackup(): string
    {
        $filename = $this->restoreId;

        return storage_path('app\public'.DIRECTORY_SEPARATOR.'backup-restore-temp'.DIRECTORY_SEPARATOR.$filename);
    }

**Reply by stefanzweifel** [reply]
Great to here it's working for you @MakaraDevOP!

Could you maybe explain or share an error message on why you needed to change the path to include the public folder?

Your local fix is not something I'm willing to make in the package. Storing the decompressed backup in `storage/app/public` would make the backup publicly available. That's a security issue if someone runs the command on a production server.

**Reply by stefanzweifel** [reply]
#40 has been merged and released in [v1.1.3](https://github.com/stefanzweifel/laravel-backup-restore/releases/tag/v1.1.3).
The path to the local decompressed backup remains in a not public-folder. So you might have to edit your vendor files again, if you update the package.

**Reply by MakaraDevOP** [reply]
I changed to a public folder because It couldn't find a file in `storage/app`, and then I changed to a public folder, and it was working well.

this function return false:

    public function hasNoDbDumpsDirectory(): bool
    {
        return ! Storage::disk($this->restoreDisk)
            ->has($this->getPathToLocalDecompressedBackup().DIRECTORY_SEPARATOR.'db-dumps');
    }

Do you have any recommend to me ?

Thanks and respect 🙏

**Reply by stefanzweifel** [reply]
Sorry for the late reply here @MakaraDevOP.

I'm not sure how I can help you here.
From your comments, I don't understand why the default location in `storage/app` is not working in your project.

Maybe the package is downloading it in the wrong location? Maybe this is also related to the fact the I couldn't test the package on a Windows machine yet.

**Reply by AshrafBasry** [reply]
I've the same issue on v1.4 
```php
$sail artisan backup:restore --backup=Incubation/only-db-2024-10-19-17-02-52.zip --disk=local --no-interaction  
                                                                                                                                                               
 Fetch list of backups from local …                                                                                                                            


 Backup downloaded to backup-restore-temp/2024-10-19-11-11-37-e1f11af4-c063-4720-aad9-957ba70f1197.zip.


 Extracted database dump from backup.


   Wnx\LaravelBackupRestore\Exceptions\NoDatabaseDumpsFound 

      "No database dumps found in backup `Incubation/only-db-2024-10-19-17-02-52.zip`."
    "Found files in db-dumps directory:"

  at vendor/wnx/laravel-backup-restore/src/Exceptions/NoDatabaseDumpsFound.php:17
     13▕     {
     14▕ 
     15▕         $files = $pendingRestore->getAvailableFilesInDbDumpsDirectory()->implode("\n");
     16▕ 
  ➜  17▕         return new static(<<<TXT
     18▕             "No database dumps found in backup `{$pendingRestore->backup}`."
     19▕             "Found files in db-dumps directory:"
     20▕             $files
     21▕         TXT

      +15 vendor frames 

  16  artisan:13
      Illuminate\Foundation\Application::handleCommand()
```

The downloaded backup.zip and the unziped files are in seperate folders
![Screenshot from 2024-10-19 21-59-04](https://github.com/user-attachments/assets/b7c89941-2777-4326-b416-47c5337f8443)

I think its because the default local disk is ('app/private') and i don't want to change it.
How to restore backups without changing the local disk path?

**Reply by stefanzweifel** [reply]
@AshrafBasry I just want to confirm: in your `config/filesystems.php`-file the config for the `local` disk does **NOT** look like this:

```php
        'local' => [
            'driver' => 'local',
            'root' => storage_path('app'),
            'throw' => false,
        ],
```

but like this?

```php
        'local' => [
            'driver' => 'local',
            'root' => storage_path('app/private'),
            'throw' => false,
        ],
```

I think the problem lies here, where I've hardcoded `app/backup-restore-temp` as a path.
https://github.com/stefanzweifel/laravel-backup-restore/blob/main/src/PendingRestore.php#L61

I have to test, if I can replicate your error and if using the `local.root`-config values solves the problem.

**Reply by stefanzweifel** [reply]
Can confirm, that this is the issue. I will push and release a fix shortly.

**Reply by AshrafBasry** [reply]
> @AshrafBasry I just want to confirm: in your `config/filesystems.php`-file the config for the `local` disk does **NOT** look like this:
> 
> ```
>         'local' => [
>             'driver' => 'local',
>             'root' => storage_path('app'),
>             'throw' => false,
>         ],
> ```
> 
> but like this?
> 
> ```
>         'local' => [
>             'driver' => 'local',
>             'root' => storage_path('app/private'),
>             'throw' => false,
>         ],
> ```
> 
> I think the problem lies here, where I've hardcoded `app/backup-restore-temp` as a path. https://github.com/stefanzweifel/laravel-backup-restore/blob/main/src/PendingRestore.php#L61
> 
> I have to test, if I can replicate your error and if using the `local.root`-config values solves the problem.

Yes it look like this
 ```
>         'local' => [
>             'driver' => 'local',
>             'root' => storage_path('app/private'),
>             'throw' => false,
>         ],
> ```

**Reply by AshrafBasry** [reply]
Waiting for the fix :+1:

**Reply by stefanzweifel** [reply]
@AshrafBasry The fix has been released. https://github.com/stefanzweifel/laravel-backup-restore/releases/tag/v1.4.2
